src/main/java/com/uid2/client/IdentityInput.java

```diff
@@ -0,0 +1,93 @@
+package com.uid2.client;
+
+import com.google.gson.JsonObject;
+
+import java.security.MessageDigest;
+
+public class IdentityInput {
+    static public IdentityInput fromEmail(String email) {
+        return new IdentityInput(IdentityType.Email, email, true);
+    }
+
+    static public IdentityInput fromPhone(String phone) {
+        return new IdentityInput(IdentityType.Phone, phone, true);
+    }
+
+    public IdentityInput withTcfConsentString(String tcf) {
```

Comment:
Is TC string actually part of identity? Would it be better to change `PublisherUid2Helper` instead? I'd also suggest wrapping TC string in a class so that it could be used in overloads and make it harder to muck things up for the caller.

Reply:
renaming IdentityInput to TokenGenerateInput